code/modules/explorer_drone/exodrone.dm

```diff
@@ -351,31 +352,36 @@ GLOBAL_LIST_EMPTY(exodrone_launchers)
 	. = ..()
 	GLOB.exodrone_launchers += src
 
-/obj/machinery/exodrone_launcher/attackby(obj/item/I, mob/living/user, params)
-	if(istype(I, /obj/item/fuel_pellet))
+/obj/machinery/exodrone_launcher/attackby(obj/item/weapon, mob/living/user, params)
+	if(istype(weapon, /obj/item/fuel_pellet))
 		if(fuel_canister)
-			to_chat(user, span_warning("There's already a fuel tank inside [src]!"))
+			to_chat(user, span_warning("There's already fuel loaded inside [src]!"))
 			return TRUE
-		if(!user.transferItemToLoc(I, src))
+		if(!user.transferItemToLoc(weapon, src))
 			return
-		fuel_canister = I
+		fuel_canister = weapon
 		update_icon()
 		return TRUE
-	else if(istype(I,/obj/item/exodrone) && user.transferItemToLoc(I, drop_location()))
+
+	if(istype(weapon, /obj/item/exodrone) && user.transferItemToLoc(weapon, drop_location()))
 		return TRUE
-	else
-		return ..()
 
-/obj/machinery/exodrone_launcher/crowbar_act(mob/living/user, obj/item/I)
-	. = ..()
-	if(fuel_canister)
-		to_chat(user, span_notice("You remove the fuel tank from [src]."))
-		fuel_canister.forceMove(drop_location())
-		fuel_canister = null
+	return ..()
+
+/obj/machinery/exodrone_launcher/crowbar_act(mob/living/user, obj/item/crowbar)
+	if(!fuel_canister)
+		return
+
+	to_chat(user, span_notice("You remove [fuel_canister] from [src]."))
+	fuel_canister.forceMove(drop_location())
+	fuel_canister = null
```

Comment:
I've recently taken to using Exited for this sort of thing, since you only need to put the removal logic in one place, and it protects against fuckups reasonably well. May not be useful for such a limited case tho